cli/debug.go

```diff
@@ -0,0 +1,142 @@
+// Copyright 2016 The Cockroach Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+// implied. See the License for the specific language governing
+// permissions and limitations under the License. See the AUTHORS file
+// for names of contributors.
+//
+// Author: Ben Darnell
+
+package cli
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+
+	"github.com/cockroachdb/cockroach/keys"
+	"github.com/cockroachdb/cockroach/roachpb"
+	"github.com/cockroachdb/cockroach/storage/engine"
+	"github.com/cockroachdb/cockroach/util/stop"
+	"github.com/spf13/cobra"
+)
+
+var debugKeysCmd = &cobra.Command{
+	Use:   "keys [directory]",
+	Short: "dump all the keys in a store",
+	Long: `
+  Pretty-prints all keys in a store.
+`,
+	RunE: runDebugKeys,
+}
+
+func openStore(cmd *cobra.Command, args []string, stopper *stop.Stopper) (engine.Engine, error) {
+	if len(args) != 1 {
+		return nil, errors.New("one argument is required")
+	}
+	dir := args[0]
+
+	db := engine.NewRocksDB(roachpb.Attributes{}, dir, 0, 0, stopper)
+	if err := db.Open(); err != nil {
+		return nil, err
+	}
+	return db, nil
+}
+
+func printKey(kv engine.MVCCKeyValue) (bool, error) {
+	fmt.Printf("%q\n", kv.Key)
+
+	return false, nil
+}
+
+func runDebugKeys(cmd *cobra.Command, args []string) error {
+	stopper := stop.NewStopper()
+	//defer stopper.Stop()
```

Comment:
Uncommented it. I was trying to debug a panic and the stopper was clobbering the stack trace.